tests/storage/datavolume.go

```diff
@@ -1141,72 +1141,57 @@ var _ = Describe(SIG("DataVolume Integration", func() {
-						PreferredStorageClassName: storageClass.Name,
-					},
+		virtualMachinePreference := &instanceType.VirtualMachinePreference{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "virtualmachinepreference-test",
+				Namespace: vm.Namespace,
+			},
+			Spec: instanceType.VirtualMachinePreferenceSpec{
+				Volumes: &instanceType.VolumePreferences{
+					PreferredStorageClassName: storageClass.Name,
 				},
+			},
+		}
+		createdPreference, err := virtClient.VirtualMachinePreference(testsuite.GetTestNamespace(nil)).Create(context.Background(), virtualMachinePreference, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(func() error {
+			err := virtClient.VirtualMachinePreference(createdPreference.Namespace).Delete(context.Background(), createdPreference.Name, metav1.DeleteOptions{})
+			if errors.IsNotFound(err) {
+				return nil
 			}
-			_, err := virtClient.VirtualMachinePreference(testsuite.GetTestNamespace(nil)).Create(context.Background(), virtualMachinePreference, metav1.CreateOptions{})
-			Expect(err).ToNot(HaveOccurred())
-
-			// Bind VirtualMachinePreference to VirtualMachine
-			vm.Spec.Preference = &v1.PreferenceMatcher{
-				Name: virtualMachinePreference.Name,
-				Kind: instancetypeapi.SingularPreferenceResourceName,
-			}
+			return err
 		})
 
-		AfterEach(func() {
-			Expect(virtClient.VirtualMachinePreference(virtualMachinePreference.Namespace).Delete(context.Background(), virtualMachinePreference.Name, metav1.DeleteOptions{})).To(Succeed())
-		})
-
-		It("should use PreferredStorageClassName when storage class not provided by VM", func() {
-			// Remove storage class name from VM definition
-			vm.Spec.DataVolumeTemplates[0].Spec.Storage.StorageClassName = nil
-
-			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm, metav1.CreateOptions{})
-			Expect(err).ToNot(HaveOccurred())
-
-			Expect(*vm.Spec.DataVolumeTemplates[0].Spec.Storage.StorageClassName).To(Equal(virtualMachinePreference.Spec.Volumes.PreferredStorageClassName))
-		})
+		// Bind VirtualMachinePreference to VirtualMachine
+		vm.Spec.Preference = &v1.PreferenceMatcher{
+			Name: virtualMachinePreference.Name,
+			Kind: instancetypeapi.SingularPreferenceResourceName,
+		}
 
-		It("should always use VM defined storage class over PreferredStorageClassName", func() {
-			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm, metav1.CreateOptions{})
-			Expect(err).ToNot(HaveOccurred())
+		vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-			Expect(*vm.Spec.DataVolumeTemplates[0].Spec.Storage.StorageClassName).NotTo(Equal(virtualMachinePreference.Spec.Volumes.PreferredStorageClassName))
-		})
+		Expect(vm.Spec.DataVolumeTemplates[0].Spec.Storage.StorageClassName).ToNot(HaveValue(BeNil()))
```

Comment:
Are you sure this is the correct assertion? it used to compare to virtualMachinePreference

Reply:
I think comparing to what ends up on the pref is the "right" way to test this (since there could be mutating in between) but obviously PreferredStorageClassName is storageClass.Name

Reply:
This is marked resolved but we still aren't comparing to the pref?